src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/SourceGenJsonTypeInfoOfT.cs

```diff
@@ -150,7 +150,8 @@ internal void AddPropertiesUsingSourceGenInfo()
                 {
                     if (hasJsonInclude)
                     {
-                        ThrowHelper.ThrowInvalidOperationException_JsonIncludeOnNonPublicInvalid(jsonPropertyInfo.ClrName!, jsonPropertyInfo.DeclaringType);
+                        Debug.Assert(jsonPropertyInfo.ClrName != null, "ClrName is not set by source gen");
```

Comment:
```suggestion
                        Debug.Assert(jsonPropertyInfo.ClrName != null, "ClrName is always set by source gen");
```

Reply:
Actually nevermind, depends on whether the string reflects the assertion or the failure condition.

Reply:
yeh, it's more like error message so says exactly opposite😄